pkg/sql/colexec/table_function/metadata_scan.go

```diff
@@ -138,113 +126,107 @@ func genRetBatch(proc process.Process, arg *Argument, metaInfos []*catalog.Metad
-
-		case catalog.MetadataScanInfoNames[catalog.META_LOC]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.META_LOC])
-
-		case catalog.MetadataScanInfoNames[catalog.DELTA_LOC]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.DELTA_LOC])
-
-		case catalog.MetadataScanInfoNames[catalog.COMMIT_TS]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.COMMIT_TS])
-
-		case catalog.MetadataScanInfoNames[catalog.SEG_ID]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.SEG_ID])
 
-		case catalog.MetadataScanInfoNames[catalog.ROWS_CNT]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.ROWS_CNT])
-
-		case catalog.MetadataScanInfoNames[catalog.NULL_CNT]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.NULL_CNT])
-
-		case catalog.MetadataScanInfoNames[catalog.COMPRESS_SIZE]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.COMPRESS_SIZE])
-
-		case catalog.MetadataScanInfoNames[catalog.ORIGIN_SIZE]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.ORIGIN_SIZE])
-
-		case catalog.MetadataScanInfoNames[catalog.MIN]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.MIN])
-
-		case catalog.MetadataScanInfoNames[catalog.MAX]:
-			retBat.Vecs[i] = vector.NewVec(catalog.MetadataScanInfoTypes[catalog.MAX])
-
-		default:
-			retBat.Clean(proc.GetMPool())
-			return nil, moerr.NewInvalidInput(proc.Ctx, "bad input select columns name %v", a)
+	for i, a := range arg.Attrs {
+		idx, ok := plan.MetadataScanInfo_MetadataScanInfoType_value[a]
+		if !ok {
+			return nil, moerr.NewInternalError(proc.Ctx, "bad input select columns name %v", a)
 		}
+
+		tp := plan2.MetadataScanColTypes[idx]
+		retBat.Vecs[i] = vector.NewVec(tp)
 	}
 
 	return retBat, nil
 }
 
-func fillMetadataInfoBat(opBat *batch.Batch, proc process.Process, arg *Argument, info *catalog.MetadataScanInfo) error {
+func fillMetadataInfoBat(opBat *batch.Batch, proc process.Process, arg *Argument, info *plan.MetadataScanInfo) error {
 	mp := proc.GetMPool()
-	for i, a := range arg.Attrs {
-		switch a {
-		case catalog.MetadataScanInfoNames[catalog.COL_NAME]:
+	for i, colname := range arg.Attrs {
+		idx, ok := plan.MetadataScanInfo_MetadataScanInfoType_value[colname]
+		if !ok {
+			opBat.Clean(proc.GetMPool())
+			return moerr.NewInternalError(proc.Ctx, "bad input select columns name %v", colname)
+		}
+
+		switch plan.MetadataScanInfo_MetadataScanInfoType(idx) {
+		case plan.MetadataScanInfo_COL_NAME:
 			vector.AppendBytes(opBat.Vecs[i], []byte(info.ColName), false, mp)
 
-		case catalog.MetadataScanInfoNames[catalog.BLOCK_ID]:
-			vector.AppendAny(opBat.Vecs[i], info.BlockId, false, mp)
+		case plan.MetadataScanInfo_BLOCK_ID:
+			var bid types.Blockid
+			if err := bid.Unmarshal(info.BlockId); err != nil {
+				return err
+			}
+			vector.AppendAny(opBat.Vecs[i], bid, false, mp)
```

Comment:
should use `AppendFixed` as much as possible, unless the second argument is indeed `any` type